lib/compiler-singlepass/src/machine.rs

```diff
@@ -447,18 +450,48 @@ impl Machine {
             }
         }
 
-        // Initialize all normal locals to zero.
-        for i in n_params..n {
-            a.emit_mov(Size::S64, Location::Imm32(0), locations[i]);
-        }
-
         // Load vmctx into R15.
         a.emit_mov(
             Size::S64,
             Self::get_param_location(0),
             Location::GPR(GPR::R15),
         );
 
+        // Stack probe.
+        //
+        // `rep stosq` writes data from low address to high address and may skip the stack guard page.
+        // so here we probe it explicitly when needed.
+        for i in (n_params..n).step_by(NATIVE_PAGE_SIZE / 8).skip(1) {
+            a.emit_mov(Size::S64, Location::Imm32(0), locations[i]);
+        }
+
+        //Initialize all normal locals to zero.
+        let mut init_stack_loc_cnt = 0;
+        let mut last_stack_loc = Location::Memory(GPR::RBP, i32::MAX);
+        for i in n_params..n {
+            match locations[i] {
+                Location::Memory(_, _) => {
+                    init_stack_loc_cnt += 1;
+                    last_stack_loc = cmp::min(last_stack_loc, locations[i]);
+                }
+                Location::GPR(_) => {
+                    a.emit_mov(Size::S64, Location::Imm32(0), locations[i]);
+                }
+                _ => unreachable!(),
+            }
+        }
+        if init_stack_loc_cnt > 0 {
+            // Since this assemblies takes up 24 bytes, If initialize more than 2 slots, These assemblies are smallar.
```

Comment:
```suggestion
            // Since these assemblies take up to 24 bytes, if more than 2 slots are initialized, then they are smaller.
```

Not sure here about the intended meaning? :-)

Reply:
thanks for corrections!

Reply:
thanks for correction! 
Suggestion is a feature I've never seen on github. I accepted it, and the commit message was not pretty, is it okay?

Reply:
That's OK yes :-)